monitor/uidmonitor/processor.go

```diff
@@ -0,0 +1,346 @@
+	err = s.processLinuxServiceStart(eventInfo, pids.(*puToPidEntry).Info)
+	return err
+
+}
+
+// Stop handles a stop event and destroy as well. Destroy does nothing for the uid monitor
+func (s *UIDProcessor) Stop(eventInfo *rpcmonitor.EventInfo) error {
+
+	contextID, err := s.generateContextID(eventInfo)
+	if err != nil {
+		return err
+	}
+
+	if contextID == triremeBaseCgroup {
+		s.netcls.Deletebasepath(contextID)
+		return nil
+	}
+	s.Lock()
+	defer s.Unlock()
+	//ignore the leading / here this is a special case for stop where i need to do a reverse lookup
+	stoppedpid := strings.TrimLeft(contextID, "/")
+	if puid, err := s.pidToPU.Get(stoppedpid); err == nil {
+		contextID = puid.(string)
+	}
+	var publishedContextID string
+	if pidlist, err := s.putoPidMap.Get(contextID); err == nil {
+		ctx := pidlist.(*puToPidEntry)
+		publishedContextID = ctx.publishedContextID
+		//Clean pid from both caches
+		delete(ctx.pidlist, stoppedpid)
+		s.pidToPU.Remove(stoppedpid)
+		if len(pidlist.(*puToPidEntry).pidlist) != 0 {
+			//Only destroy the pid that is being stopped
+			s.netcls.DeleteCgroup(stoppedpid)
+			return nil
+		}
+		//We are the last here lets send stop
+		if err = s.puHandler.HandlePUEvent(publishedContextID, monitor.EventStop); err != nil {
+			zap.L().Warn("Failed to stop trireme PU ",
+				zap.String("contextID", contextID),
+				zap.Error(err),
+			)
+		}
+		s.putoPidMap.Remove(contextID)
+		if err = s.contextStore.RemoveContext(contextID); err != nil {
+			zap.L().Error("Failed to clean cache while destroying process",
+				zap.String("contextID", contextID),
+				zap.Error(err),
+			)
+		}
+		if err = s.puHandler.HandlePUEvent(publishedContextID, monitor.EventDestroy); err != nil {
+			zap.L().Warn("Failed to Destroy clean trireme ",
+				zap.String("contextID", contextID),
+				zap.Error(err),
+			)
+		}
+
+		s.netcls.DeleteCgroup(stoppedpid)
+
+	}
+
+	return nil
+
+}
+
+// Destroy handles a destroy event
+func (s *UIDProcessor) Destroy(eventInfo *rpcmonitor.EventInfo) error {
+	//Destroy is not used for the UIDMonitor since we will destroy when we get stop
+	//This is to try and save some time .Stop/Destroy is two RPC calls.
+	//We don't define pause on uid monitor so stop is always followed by destroy
+	return nil
+
+}
+
+// Pause handles a pause event
+func (s *UIDProcessor) Pause(eventInfo *rpcmonitor.EventInfo) error {
+
+	contextID, err := s.generateContextID(eventInfo)
+	if err != nil {
+		return fmt.Errorf("Couldn't generate a contextID: %s", err)
+	}
+
+	return s.puHandler.HandlePUEvent(contextID, monitor.EventPause)
+}
+
+// ReSync resyncs with all the existing services that were there before we start
+func (s *UIDProcessor) ReSync(e *rpcmonitor.EventInfo) error {
+
+	deleted := []string{}
+	reacquired := []string{}
+	marktoPID := map[string][]string{}
+	defer func() {
+		if len(deleted) > 0 {
+			zap.L().Info("Deleted dead contexts", zap.String("Context List", strings.Join(deleted, ",")))
```

Comment:
in zap.X() the key is translated to json. json keys with space are bad. Also, let zap do the job for you:

```go
zap.L().Info("Deleted dead contexts", zap.Strings("contexts", deleted))
```